lib/api/CDataFrameAnalyzer.cc

```diff
@@ -131,6 +131,9 @@ void CDataFrameAnalyzer::run() {
         auto& instrumentation = analysisRunner->instrumentation();
         CDataFrameAnalysisInstrumentation::CScopeSetOutputStream setStream{
             instrumentation, *outStream};
+        instrumentation.updateMemoryUsage(
+            static_cast<std::int64_t>(m_DataFrame->memoryUsage()));
+        instrumentation.flush();
```

Comment:
Since we're accounting for the frame's memory usage here we need to update to only consider the delta when resizing later. For example at [this](https://github.com/elastic/ml-cpp/blob/master/lib/maths/CBoostedTreeFactory.cc#L280) line you'd need to do something like:
```
std::size_t numberLossParameters{m_TreeImpl->m_Loss->numberParameters()};
std::size_t lastMemoryUsage{core::CMemory::dynamicSize(frame)};
m_TreeImpl->m_ExtraColumns = frame.resizeColumns(
    m_TreeImpl->m_NumberThreads, extraColumns(numberLossParameters));
m_TreeImpl->m_Instrumentation->updateMemoryUsage(
    core::CMemory::dynamicSize(frame) - lastMemoryUsage);
```
and similar changes in `COutliers.cc`.

Reply:
Good catch! :+1: I fixed double counting in fa785e2